django/db/models/query.py

```diff
@@ -1137,6 +1130,21 @@ def _has_filters(self):
         """
         return self.query.has_filters()
 
+    def _check_allowed(self, action):
+        require_no_limits = {
```

Comment:
`order_by` only needs to check for slice thus it can call this method like
`check_queryset_method_allowed('order_by', combinator_check=False)`